torchgeo/datasets/geo.py

```diff
@@ -418,6 +420,61 @@ def __init__(
         self._crs = cast(CRS, crs)
         self._res = cast(float, res)
 
+    def list_files(self, filename_glob: Optional[str] = None) -> list[str]:
```

Comment:
Can we remove the `filename_glob` parameter for now? This is also being discussed in #1524 and I don't think we've settled on how to handle this yet.

Reply:
Also wondering if we should change this to:
```python
@propery
def files(self) -> list[str]:
    ...
```
I think `dataset.files` feels cleaner than `dataset.list_files()`

Reply:
I like the idea of a property, but it should still re-create the list every time it is called, as the file structure may change, do you agree?

Reply:
I included the possibility to override filename_glob only in order to easier support the code that [looks for zip-files](https://github.com/microsoft/torchgeo/blob/a5d3d6ef9b26690bb632e76ab123cef9466139c9/tests/datasets/test_eudem.py#L36). If we can still solve that, I am happy to remove it.

Reply:
I think we can remove that and use what we previously used to find zip files, just using glob.

I'm fine with re-creating the list every time. If it becomes slow or we need to do it multiple times we can cache it.